src/Spectrometer-OceanOptics_OceanDirect/main.py

```diff
@@ -0,0 +1,325 @@
+        self.od = self.restore_parameter(self.unique_ocean_direct_api_identifier)
+        if self.od is None:
+            self.od = OceanDirectAPI.OceanDirectAPI()
+            self.store_parameter(self.unique_ocean_direct_api_identifier, self.od)
+
+        number_devices = self.od.find_usb_devices()
+        # print("Found devices:", number_devices)
+
+        dev_ids = self.od.get_device_ids()
+
+        # print("Device IDs:", dev_ids)
+
+        ports = []
+       
+        for dev_id in dev_ids:
+
+            spectrometer = self.od.open_device(dev_id)
+            serial_number = spectrometer.get_serial_number()
+            model = spectrometer.get_model()
+
+            port_serial_number = model + " - " + serial_number
+            ports.append(port_serial_number)
+
+            self.unique_spectrometer_identifier = "OceanDirectAPI_spectrometer_" + port_serial_number
+            self.store_parameter(self.unique_spectrometer_identifier, spectrometer)
+
+            # spectrometer.close_device()
+            # od.close_device(dev_id)
+
+        # print("Ports:", ports)
+
+        if len(ports) == 0:
+            self.message_Box(
+                "No spectrometer found! Please make sure it is connected and the OceanDirect API is installed.")
+
+        # returns a list of ports
+        return ports
+
+    # Remains for later in case calibration will be added
+    """
+    def get_CalibrationFile_properties(self, port):
+        # returns two string
+        # 1 file ending
+        # 2 string in file_name
+
+        # Example: USB4F05021_040309.IrradCal
+
+        serialno = str(port[str(port).find(":") + 1:-1])
+
+        if serialno == "":
+            serialno = "noserialnumber"
+
+        return [".IrradCal", serialno]
+    """
+
+    def set_GUIparameter(self):
+        gui_parameter = {
+            "SweepMode": ["None", "Integration time in s"],
+            # "IntegrationTimeAutomatic": False,
+            "IntegrationTime": 0.1,
+            "Average": 1,
+            "Trigger": list(self.trigger_modes.keys()),
+            "TriggerDelay": 0.0,
+        }
+
+        return gui_parameter
+
+    def get_GUIparameter(self, parameter={}):
+        self.integration_time = parameter["IntegrationTime"]
+        # self.automatic = parameter.get("IntegrationTimeAutomatic", False)
+        self.integration_time_automatic_max = parameter.get("IntegrationTimeMax", 10.0)
+        self.sweep_mode = parameter["SweepMode"]
+        self.average = parameter["Average"]
+        self.port_serial_number = parameter["Port"]
+        self.calibration = parameter.get("Calibration", "")
+        self.trigger_mode = parameter["Trigger"]
+        self.trigger_delay = parameter.get("TriggerDelay", 0.0)
+        
+        if self.calibration == "" or self.calibration == "None":
+            self.units = ["nm", "", "s"]
+        else:
+            self.units = ["nm", "µJ", "s"]  # TODO: check unit intensity according to calibration
```

Comment:
Is this still unchecked? Who will check it when?
-> It might be useful to mention this as a limitation somewhere.

Reply:
changed it